File: src/main/scala/com/vc/medium/L776.scala
package com.vc.medium

class TreeNode776(var _value: Int) {
  var value: Int = _value
  var left: TreeNode776 = null
  var right: TreeNode776 = null
}

object L776 {
  def splitBST(root: TreeNode776, V: Int): Array[TreeNode776] = {
    if(root == null) return Array(null, null)
    if(root.value <= V) {
      val binarySplit = splitBST(root.right, V)
      root.right = binarySplit(0)
      binarySplit(0) = root
      binarySplit
    }
    else {
      val binarySplit = splitBST(root.left, V)
      root.left = binarySplit(1)
      binarySplit(1) = root
      binarySplit
    }
  }
}

File: src/main/java/com/vc/medium/L776.java
package com.vc.medium;

import com.vc.hard.TreeNode;

class L776 {
    public TreeNode[] splitBST(TreeNode root, int V) {
        if(root == null) return new TreeNode[2];

        if(root.val > V) {
            TreeNode[] res = splitBST(root.left, V);
            root.left = res[1];
            return new TreeNode[]{res[0], root};
        }
        else {
            TreeNode[] res = splitBST(root.right, V);
            root.right = res[0];
            return new TreeNode[]{root, res[1]};
        }
    }
}
